mod/admin.php

```diff
@@ -290,7 +290,7 @@ function admin_page_federation(&$a) {
 		// what versions for that platform do we know at all?
 		// again only the active nodes
 		$v = qu('SELECT COUNT(*) AS `total`, `version` FROM `gserver`
-				WHERE `last_contact` > `last_failure` AND `platform` LIKE "%s"  AND `version` != ""
+				WHERE `last_contact` > `last_failure` AND `platform` LIKE "%s"  AND `version` RLIKE "^[[:digit:]]+[-.][[:digit:]]+"
```

Comment:
I'm not totally happy with this. We should sanitize the data when we store it - but not when we fetch it.